website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,799 @@
+        }
+        else if (self.type === 'node') {
+            return $osf.apiV2Url('nodes/' + self.data.uid + '/children/', { query : { 'related_counts' : true, 'embed' : 'contributors' }});
+        }
+        // If nothing
+        throw new Error('Link could not be generated from linkObject data');
+    };
+    self.link = self.generateLink();
+};
+
+if (!window.fileBrowserCounter) {
+    window.fileBrowserCounter = 0;
+}
+function getUID() {
+    window.fileBrowserCounter = window.fileBrowserCounter + 1;
+    return window.fileBrowserCounter;
+}
+
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    xhr.setRequestHeader('Accept', 'application/vnd.api+json; ext=bulk');
+
+};
+
+var xhrconfigBulk = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader("Content-Type", "application/vnd.api+json; ext=bulk");
+    xhr.setRequestHeader("Accept", "application/vnd.api+json; ext=bulk");
+};
+
+/**
+ * Initialize File Browser. Prepeares an option object within FileBrowser
+ * @constructor
+ */
+var FileBrowser = {
+    controller : function (options) {
+        var self = this;
+        self.wrapperSelector = options.wrapperSelector;  // For encapsulating each implementation of this component in multiple use
+        self.currentLink = ''; // Save the link to compare if a new link is being requested and avoid multiple calls
+        self.reload = m.prop(false); // Gets set to true when treebeard link changes and it needs to be redrawn
+        self.nonLoadTemplate = m.prop(m('.fb-non-load-template.m-md.p-md.osf-box', 'Loading...')); // Template for when data is not available or error happens
+
+        // VIEW STATES
+        self.showInfo = m.prop(true); // Show the info panel
+        self.showSidebar = m.prop(true); // Show the links with collections etc. used in narrow views
+        self.showCollectionMenu = m.prop(false); // Show hide ellipsis menu for collections
+        self.collectionMenuObject = m.prop(); // Collection object to complete actions on menu
+        self.resetCollectionMenu = function () {
+            self.collectionMenuObject({item : {label:null}, x : 0, y : 0});
+        };
+
+        // Default system collections
+        self.collections = [
+            new LinkObject('collection', { path : 'users/me/nodes/', query : { 'related_counts' : true, 'page[size]'  : 100, 'embed' : 'contributors' }, systemCollection : true}, 'All My Projects'),
+            new LinkObject('collection', { path : 'registrations/', query : { 'related_counts' : true, 'page[size]'  : 100, 'embed' : 'contributors'}, systemCollection : true}, 'All My Registrations'),
+        ];
+
+        var collectionsUrl = $osf.apiV2Url('collections/', { query : {'related_counts' : true, 'sort' : 'date_created'}});
+        m.request({method : 'GET', url : collectionsUrl, config : xhrconfig}).then(function(result){
+            console.log(result);
```

Comment:
Hmm, nothing like console logging things... Okay I will remove these before sending out to final review.